filebeat/input/filestream/fswatch.go

```diff
@@ -496,7 +508,7 @@ func (s *fileScanner) toFileDescriptor(it *ingestTarget) (fd loginp.FileDescript
 			return fd, fmt.Errorf("failed to read %d bytes from %q to compute fingerprint, read only %d", written, fd.Filename, s.cfg.Fingerprint.Length)
 		}
 
-		fd.Fingerprint = hex.EncodeToString(s.hasher.Sum(nil))
+		fd.Fingerprint = fmt.Sprintf("%x", s.hasher.Sum64())
```

Comment:
This simple benchmark tells me that `fmt.Sprintf("%x")` is slower but perhaps more efficient on memory:

```go
package main

import (
	"crypto/rand"
	"encoding/hex"
	"fmt"
	"testing"
)

const size = 64

func BenchmarkMain(b *testing.B) {
	data := make([]byte, size)
	_, _ = rand.Read(data)

	var val1, val2 string

	b.Run("hex", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			val1 = hex.EncodeToString(data)
			if len(val1) == 0 {
				b.FailNow()
			}
		}
	})

	b.Run("sprintf", func(b *testing.B) {
		for i := 0; i < b.N; i++ {
			val2 = fmt.Sprintf("%x", data)
			if len(val2) == 0 {
				b.FailNow()
			}
		}
	})

	if val1 != val2 {
		b.FailNow()
	}
}
```

```
BenchmarkMain/hex-10            11652643               101.0 ns/op           256 B/op          2 allocs/op
BenchmarkMain/sprintf-10         7365199               162.8 ns/op           152 B/op          2 allocs/op
```

Reply:
So the reason for this change this make it even slightly better. So if you see I used Sum64 instead of Sum which returns different types.

`hex.EncodeToString` expects a slice but `Sum64` returns `uint64` so it was necessary to use `fmt.Sprintf("%x")`

Both in the end give the same result. The main difference is inside Sum and Sum64's implementation. In Sum a new buffer is allocated as nil is passed and inside [8]byte is also created along with encoding it to BigEndian but in the same machine I don't think it is required. So better choice was to use Sum64.